pkg/framework/plugins/podlifetime/pod_lifetime.go

```diff
@@ -131,3 +132,15 @@ func (d *PodLifeTime) Deschedule(ctx context.Context, nodes []*v1.Node) *framewo
 
 	return nil
 }
+
+func init() {
```

Comment:
I like have the registry implicitly ([pluginbuilder.go](https://github.com/kubernetes-sigs/descheduler/pull/1003/files#diff-907243a6cfbb3c6f27a53916054eaa5b626cfda6ab368aad8f528769a3bb7451R38)) initialized with `init()` (because it is part of the framework internals), but I'm thinking the plugins should have an explicit function call to register themselves. wdyt?

Reply:
I think that makes sense as well. But I also like plugins that automatically register themselves if you import them, like I did here.

Using a plugin would then just require a '_' import in a register.go file or something like that. (this makes it super simple for people to use a plugin and only have to change a single file). Not using a plugin means just deleting a line from that list of imports. This is specially interesting when thinking of out-of-tree plugins.

register.go would look like:
```
// packages imported here are registered to the pluginBuilder Plugin Registry.
//nolint:revive
import (
	_ "sigs.k8s.io/descheduler/pkg/framework/plugins/removepodsviolatinginterpodantiaffinity"
	_ "sigs.k8s.io/descheduler-external-plugins/pkg/plugins/custompluginexample"
)
```

Reply:
Yeah, it's cool for sure. And it makes sense to use it for the internals. But does it provide value beyond that convenience? Even then, it's only saving 1 line.

I think the user-facing functions should be explicit rather than automatically hooking into framework internals as soon as they're imported. Doing so makes it safer (compile time checks on unused plugins + attempts to use unimported plugins) and clearer about what's going on with less magic for the user to understand.

Reply:
I think it does more than save one line. Having a new explicit Register function, or even using a similar approach to the KS with the WithPlugin wrapper functions adds complexity to how this works, and I think it works against the UX of the user.

> compile time checks on unused plugins + attempts to use unimported plugins

I don't see how that helps with compile time checks on unused plugins, and attempts to use unimported plugins would trigger similar errors, and we can still treat and have meaningful logs either way.

To me it is as explicit as having a function call, importing or not importing a new plugin. I might be missing something about specifics of the implementation

Reply:
> To me it is as explicit as having a function call, importing or not importing a new plugin.

A function to manually register the plugin is much more explicit, and it makes the registration flow easier to follow and debug. Implicitly registering the plugin by importing it means you have to be vigilant about where you import that plugin (for example if 2 plugins share functionality).

Auto-registration is something that could be added later, at a time when we are more comfortable with the framework. Adding it as a day-0 feature is too much in my opinion. I would like to see an end-to-end product first before adding convenience functionality like this.

Reply:
I see your point! Gonna add the Register func and register all internal plugins by default :)